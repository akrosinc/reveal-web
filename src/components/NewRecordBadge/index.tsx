import moment from 'moment';
import React from 'react';
import { Badge } from 'reactstrap';
import { NEW_TITLE } from '../../configs/lang';

/** interface for NewRecordBadge props */
interface NewRecordBadgeProps {
  className: string;
  color: string;
  numDays: number;
  pill: boolean;
  recordDate: string | Date | null;
  text: string;
}

/** Simple component that displays a badge for new records */
const NewRecordBadge = (props: NewRecordBadgeProps) => {
  const { className, color, numDays, pill, recordDate, text } = props;
  if (
    recordDate &&
    moment(recordDate) &&
    moment(recordDate).isAfter(moment().subtract(numDays, 'days'))
  ) {
    return (
      <Badge color={color} pill={pill} className={className}>
        {text}
      </Badge>
    );
  }
  return null;
};

/** define default props */
const defaultProps: NewRecordBadgeProps = {
  className: 'new-record',
  color: 'warning',
  numDays: 1,
  pill: true,
  recordDate: null,
  text: NEW_TITLE,
};

/** declare default props for NewRecordBadge */
NewRecordBadge.defaultProps = defaultProps;

export default NewRecordBadge;
