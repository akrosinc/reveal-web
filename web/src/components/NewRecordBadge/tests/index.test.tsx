import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import moment from 'moment';
import React from 'react';
import NewRecordBadge from '..';

describe('components/NewRecordBadge', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(<NewRecordBadge />);
  });

  it('renders badge correctly', () => {
    const wrapper = mount(<NewRecordBadge />);
    // badge not rendered if recordDate not supplied
    expect(wrapper.find('Badge').length).toEqual(0);

    // renders okay for a date within threshold
    const wrapper2 = mount(
      <NewRecordBadge
        recordDate={moment()
          .subtract(1, 'hour')
          .toDate()}
      />
    );
    expect(wrapper2.find('Badge').length).toEqual(1);
    expect(toJson(wrapper2.find('Badge'))).toMatchSnapshot('badge');

    // does not render badhe for a far past date
    const wrapper3 = mount(
      <NewRecordBadge
        recordDate={moment()
          .subtract(10, 'days')
          .toDate()}
      />
    );
    expect(wrapper3.find('Badge').length).toEqual(0);
  });

  it('renders okay with custom props', () => {
    // renders okay for a date within threshold
    const wrapper = mount(
      <NewRecordBadge
        className="mosh"
        color="danger"
        numDays={2}
        pill={false}
        recordDate={moment()
          .subtract(26, 'hours')
          .toDate()}
        text="Cool"
      />
    );
    expect(wrapper.find('Badge').length).toEqual(1);
    expect(toJson(wrapper.find('Badge'))).toMatchSnapshot('badge');
  });
});
