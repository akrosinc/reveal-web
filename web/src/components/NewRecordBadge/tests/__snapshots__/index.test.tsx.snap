// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NewRecordBadge renders badge correctly: badge 1`] = `
<Badge
  className="new-record"
  color="warning"
  pill={true}
  tag="span"
>
  <span
    className="new-record badge badge-warning badge-pill"
  >
    New
  </span>
</Badge>
`;

exports[`components/NewRecordBadge renders okay with custom props: badge 1`] = `
<Badge
  className="mosh"
  color="danger"
  pill={false}
  tag="span"
>
  <span
    className="mosh badge badge-danger"
  >
    Cool
  </span>
</Badge>
`;
